Fixes the error:

http.c:1921:17: error: member reference base type 'void' is not a structure or 
union
                if (ftruncate(fileno(result), 0) < 0) {
                              ^~~~~~~~~~~~~~
/usr/include/stdio.h:410:36: note: expanded from macro 'fileno'
#define fileno(p)       (!__isthreaded ? __sfileno(p) : (fileno)(p))
                                         ^~~~~~~~~~~~
/usr/include/stdio.h:401:26: note: expanded from macro '__sfileno'
#define __sfileno(p)    ((p)->_file)

1 error generated.

--- /usr/ports/pobj/git-2.35.4/git-2.35.4/http.c.orig	Mon Aug  1 06:50:19 2022
+++ /usr/ports/pobj/git-2.35.4/git-2.35.4/http.c	Mon Aug  1 06:50:31 2022
@@ -1918,7 +1918,7 @@
 			return HTTP_START_FAILED;
 		}
 		rewind(result);
-		if (ftruncate(fileno(result), 0) < 0) {
+		if (ftruncate(fileno((FILE *)result), 0) < 0) {
 			error_errno("unable to truncate a file");
 			return HTTP_START_FAILED;
 		}
